src/FGame.cpp

```diff
@@ -89,6 +89,8 @@ void FGame::processEvents()
 			
 		if (sf::Keyboard::isKeyPressed(sf::Keyboard::Space))
 			it->fire();
+		else if (!sf::Keyboard::isKeyPressed(sf::Keyboard::Space)) 
+			it->reload(m_engine->getTickCount());
```

Comment:
This is a bit overkill, as it means the rendering engine is responsible for reloading the weapons in the game engine.